### PR TITLE
Convert argument parsing to use argparse4j

### PR DESCRIPTION
**Pull request by grimreaper**
This makes the code a little more obvious as to what's happening and
provides usable --help output.

### CODE REVIEW

**Comment by cpmeister** [comment]
Other than some formatting inconsistencies, this looks great. Fix the tab formatting and I'll merge it.

**Comment by cpmeister** [comment]
Also, I'd like you to exclude the gradle-wrapper.properties changes from you commit. It isn't relevant to this PR and allowing might cause merging issues for other developers.

**Comment by grimreaper** [comment]
I figured out why the gradle wrapper task was being run, and fixed it. Will fix formatting soon.

**Comment by grimreaper** [comment]
formatting changed

File: code/src/java/pcgen/system/Main.java [hunk 1]
```diff
@@ -192,176 +171,34 @@ public static boolean loadCharacterAndExport(String characterFile, String export
...
+		startGMGen = args.getBoolean("gmgen");
+		startNPCGen = args.getBoolean("npc");
+		ignoreJavaVer = args.getBoolean("J");
+		settingsDir = args.getString("settingsdir");
+		campaignMode = args.getString("campaignmode");
+        characterSheet = args.get("D");
+        exportSheet = args.get("E");
+        partyFile = args.get("p");
+        characterFile = args.get("c");
+        outputFile = args.get("o");
```
Comment (cpmeister) [inline code/src/java/pcgen/system/Main.java]: This formatting needs to be changed. It is using spaces instead of tabs.

Comment (grimreaper) [inline code/src/java/pcgen/system/Main.java]: fixed spacing and IDE settings to avoid this in the future. Thanks!

**Pull request merged**